operations/security_and_authentication/Add_Root_Service_Account_for_Gigabyte_Controllers.md

```diff
@@ -4,87 +4,159 @@ By default, Gigabyte BMC and CMC controllers have the `admin` service
-   ipmitool -U <user> -P <password> -I lanplus -H <ip> channel setaccess 1 4 callin=on ipmi=on link=on
-   Set User Access (channel 1 id 4) successful.
-   ```
-
-1. Additionally, if the target controller is a BMC and not a CMC, run the following command:
-
-   ```bash
-   ipmitool -U <user> -P <password> -I lanplus -H <ip> sol payload enable 1 4
-   ```
-
-1. Verify that the `root` service account is now configured.
-
-   ```bash
-   curl -s -k -u <user>:<password> https://<xname>/redfish/v1/AccountService/Accounts | jq ".Members"
-   ```
-
-   ```json
-   [
-     {
-       "@odata.id": "/redfish/v1/AccountService/Accounts/4"
-     },
-     {
-       "@odata.id": "/redfish/v1/AccountService/Accounts/1"
-     }
-   ]
-   ```
-   ```bash
-   curl -s -k -u <user>:<password> https://<xname>/redfish/v1/AccountService/Accounts/4 | jq '. | { Name: .Name, UserName: .UserName, RoleId: .RoleId }'
-   ```
-
-   ```json
-   {
-     "Name": "root",
-     "UserName": "root",
-     "RoleId": "Administrator"
-   }
-   ```
+## Prerequisites
+
+- The BMC is accessible over the network via hostname or IP address.
+
+## Procedure
+
+1. (`ncn#`) Retrieve the root user password for this BMC.
+
+    - **If configuring a BMC already present in the system**, then retrieve the device-specific root user password from Vault.
+
+        ```bash
+        BMC=x3000c0s3b0
+        EXPECTED_ROOT_PASSWORD=$(cray scsd bmc creds list --targets $BMC --format json | jq .Targets[].Password -r)
+        ```
+
+        The following output indicates that Vault does not contain a device-specific root user password for the specified BMC. In that case, use the system default air-cooled BMC root password described in the step below.
+
+        ```text
+        jq: error (at <stdin>:3): Cannot iterate over null (null)
+        ```
+
+    - **If configuring a new BMC being added to the system**, then retrieve the system's default air-cooled BMC root user password from Vault.
+
+        ```bash
+        VAULT_PASSWD=$(kubectl -n vault get secrets cray-vault-unseal-keys -o json |
+                            jq -r '.data["vault-root"]' |  base64 -d)
+        EXPECTED_ROOT_PASSWORD=$(kubectl -n vault exec -it cray-vault-0 -c vault -- env \
+            VAULT_TOKEN="${VAULT_PASSWD}" VAULT_ADDR=http://127.0.0.1:8200 VAULT_FORMAT=json \
+            vault kv get secret/reds-creds/defaults | jq .data.Cray.password -r)
+        ```
+
+1. (`ncn#`) If desired, verify the contents of `EXPECTED_ROOT_PASSWORD`.
+
+    ```bash
+    echo $EXPECTED_ROOT_PASSWORD
+    ```
+
+1. (`ncn#`) Set an environment variable containing the hostname or current IP address of the BMC. If coming from the [Add Worker, Storage, or Master NCNs](../node_management/Add_Remove_Replace_NCNs.md#add-worker-storage-master)
+    procedure, then the IP address should already be stored in the `BMC_IP` environment variable.
+
+    Via hostname:
+
+    ```bash
+    export BMC=x3000c0s3b0
```

Comment:
Is the export necessary? I don't see a case where we're starting some subprocess that needs to access it. I only see it being used on the command lines here.
```suggestion
    BMC=x3000c0s3b0
```